metrics/darwin/interface.go

```diff
@@ -59,73 +45,26 @@ func (g *InterfaceGenerator) Generate() (metrics.Values, error) {
 	for name, value := range prevValues {
 		currValue, ok := currValues[name]
 		if ok {
-			ret[name+".delta"] = (currValue - value) / g.Interval.Seconds()
+			ret[name+".delta"] = float64(currValue-value) / g.Interval.Seconds()
 		}
 	}
 
 	return metrics.Values(ret), nil
 }
 
-func (g *InterfaceGenerator) collectIntarfacesValues() (metrics.Values, error) {
-	out, err := exec.Command("netstat", "-bni").Output()
+func (g *InterfaceGenerator) collectIntarfacesValues() (map[string]uint64, error) {
+	networks, err := network.Get()
 	if err != nil {
-		interfaceLogger.Errorf("Failed to invoke netstat: %s", err)
+		interfaceLogger.Errorf("failed to get network statistics: %s", err)
 		return nil, err
 	}
-
-	lineScanner := bufio.NewScanner(bytes.NewReader(out))
-
-	results := make(map[string]float64)
-	hasNonZeroValue := false
-	for lineScanner.Scan() {
-		line := lineScanner.Text()
-		fields := strings.Fields(line)
-		name := util.SanitizeMetricKey(regexp.MustCompile(`\*`).ReplaceAllString(fields[0], ""))
-		if match, _ := regexp.MatchString(`^lo\d+$`, name); match {
-			continue
-		}
-		if match, _ := regexp.MatchString(`^<Link#\d+>$`, fields[2]); match {
-			rxIndex, txIndex := getFieldIndex(fields)
-			rxKey := fmt.Sprintf("interface.%s.rxBytes", name)
-			rxValue, rxErr := strconv.ParseFloat(fields[rxIndex], 64)
-			if rxErr != nil {
-				interfaceLogger.Warningf("Failed to parse host interfaces: %s", err)
-				break
-			}
-			results[rxKey] = rxValue
-			if rxValue != 0 {
-				hasNonZeroValue = true
-			}
-			txKey := fmt.Sprintf("interface.%s.txBytes", name)
-			txValue, txErr := strconv.ParseFloat(fields[txIndex], 64)
-			if txErr != nil {
-				interfaceLogger.Warningf("Failed to parse host interfaces: %s", err)
-				break
-			}
-			results[txKey] = txValue
-			if txValue != 0 {
-				hasNonZeroValue = true
-			}
-		}
-	}
-	// results (eth0) sample
-	/** [%!s(*metrics.Value=&{interface.eth0.rxBytes 6.6074069281e+10})
-	     %!s(*metrics.Value=&{interface.eth0.txBytes 9.180531994e+09})
-	    ]
-	**/
-	if hasNonZeroValue {
-		return metrics.Values(results), nil
+	if len(networks) == 0 {
+		return nil, nil
```

Comment:
Does it make difference between returning empty map and returning nil?

Reply:
No, nothing crashes.
```go
gore> x := func() map[string]uint64 { return nil }()
map[string]uint64(nil)
gore> x
map[string]uint64(nil)
gore> for _, name := range x { println(name) }
gore> x["interface.foo.rxBytes"]
0x0
gore> currValue, ok := x["interface.foo.rxBytes"]
0x0
false
```